pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -683,7 +667,47 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongos(cr *api.PerconaServerMon
 	if deplSpec.Template.Annotations == nil {
 		deplSpec.Template.Annotations = make(map[string]string)
 	}
-	for k, v := range annotations {
+
+	if len(secretAnnotations) == 0 {
+		// we shoud check if sfs secrets were changes
+		// if yes - update mongos secret hash also
+
+		sts, err := r.getCfgStatefulset(cr)
+		if err != nil {
+			return errors.Wrap(err, "failed to get cfg ststefulset")
+		}
+
+		cfgRsSecretHash := ""
+		cfgRsSecretHashTS := ""
+		if v, ok := sts.Spec.Template.Annotations["last-applied-secret"]; ok {
+			cfgRsSecretHash = v
```

Comment:
why we need this check?
you don't check then this variable if it's empty, so you can assign a value from a map directly.

Reply:
Thanks
https://github.com/percona/percona-server-mongodb-operator/pull/537/commits/9ee1d535a27269124c8aebc7462672f31337533b